ibis/expr/api.py

```diff
@@ -1884,7 +1884,9 @@ def between_time(arg, lower, upper, timezone=None):
     """
 
     if isinstance(arg.op(), _ops.Time):
-        arg = arg.op().args[0].cast(dt.Timestamp(timezone=timezone))
+        arg = arg.op().args[0]
```

Comment:
maybe a commento about why this is needed here?

Reply:
Done